tests/test_mpi.py

```diff
@@ -2727,6 +2911,71 @@ def test_adjoint_F_no_omp(self, mode):
         self.run_adjoint_F(3)
 
 
+class TestElastic:
+
+    @pytest.mark.parallel(mode=[(1, 'diag')])
+    def test_elastic_structure(self, mode):
+
+        so = 4
+        grid = Grid(shape=(3, 3))
+
+        v = VectorTimeFunction(name='v', grid=grid, space_order=so)
+        tau = TensorTimeFunction(name='t', grid=grid, space_order=so)
+
+        damp = Function(name='damp', grid=grid)
+        l = Function(name='lam', grid=grid)
+        mu = Function(name='mu', grid=grid)
+        ro = Function(name='b', grid=grid)
+
+        # The receiver
+        rec = SparseTimeFunction(name="rec", grid=grid, npoint=1, nt=10)
+        rec_term = rec.interpolate(expr=v[0] + v[1])
+
+        # First order elastic wave equation
+        pde_v = v.dt - ro * div(tau)
+        pde_tau = (tau.dt - l * diag(div(v.forward)) -
+                   mu * (grad(v.forward) + grad(v.forward).transpose(inner=False)))
+
+        # Time update
+        u_v = Eq(v.forward, damp * solve(pde_v, v.forward))
+        u_t = Eq(tau.forward, damp * solve(pde_tau, tau.forward))
+
+        op = Operator([u_v] + [u_t] + rec_term)
+        assert len(op._func_table) == 11
+
+        calls = [i for i in FindNodes(Call).visit(op) if isinstance(i, HaloUpdateCall)]
+
+        # The correct we want
+        assert len(calls) == 5
+
+        assert len(FindNodes(HaloUpdateCall).visit(op.body.body[1].body[1].body[0])) == 1
```

Comment:
are you saying these are not covered already in the new `test_issue_*` that you've added above?

it seems to be the same dependency pattern, hence a redundant test...